utils/http/errors.py

```diff
@@ -0,0 +1,17 @@
+class HttpErrorResponseError(Exception):
+    """Generic Response Error exception
+
+    It helps to stop processing a view function by raising an exception.
+    It is like `Http404` exception but with ability to define your response.
+
+    Status code defaults to 404 but can be overridden.
+
+    NOTE: `htk.middleware.classes.HttpErrorResponseMiddleware` MUST be in
+    MIDDLEWARES in Django Settings.
+    """
+
+    def __init__(self, response, status_code=None):
+        self.response = response
+        self.response.status_code = (
+            status_code if status_code is not None else 404
```

Comment:
I may have been wrong on this, depending on the answer to my question above.

In any case, if this code is correct locally, then the preference would be to change `404` to be the default value of `status_code` on the function parameters, and change this to just be `self.response.status_code = status_code`

Reply:
Updated this code. You might want to take another look at this. I think it is better now.